beacon-chain/blockchain/process_block.go

```diff
@@ -269,15 +271,16 @@ func (s *Service) onBlockBatch(ctx context.Context, blks []*ethpb.SignedBeaconBl
 	if !verify {
 		return nil, nil, errors.New("batch block signature verification failed")
 	}
-	for r, st := range boundaries {
-		if err := s.stateGen.SaveState(ctx, r, st); err != nil {
+	for r, st := range boundaryStates {
+		if err := s.stateGen.SaveState(ctx, r, boundaryStateRoots[r], st); err != nil {
```

Comment:
Why is this a loop into saving the state? No `saveStates`?

Reply:
This was from before, I didn't want to touch the original implementation and get out of the scope. We can look into optimizing this later : )